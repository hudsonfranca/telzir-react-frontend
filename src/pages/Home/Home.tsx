import React, { useRef, useState, useEffect } from 'react';
import gsap from 'gsap';
import { useIntersection } from 'react-use';

import { Parallax } from 'react-parallax';

import { Formik, FormikHelpers } from 'formik';
import * as Yup from 'yup';
import {
  SectionOne,
  Container,
  PlansSection,
  Card,
  Title,
  CardTitle,
  CardBody,
  Label,
  Form,
  Input,
  Text,
  Select,
  InputTitle,
  ButtonSubmit,
} from './styles';
import Image1 from '../../assets/home.jpg';
import phone from '../../assets/phone.jpg';
import phone2 from '../../assets/phone2.jpg';
import api from '../../services/api';

const validate = Yup.object().shape({
  minutes: Yup.number().required('Por favor forneça os minutos '),
  planId: Yup.number().required('Por favor selecione um plano'),
  priceId: Yup.number().required('Por favor selecione uma horigem e destino'),
});

interface PriceProps {
  id: number;
  price: number;
  createdAt: string;
  updatedAt: string;
  source: {
    id: number;
    code: number;
    createdAt: string;
    updatedAt: string;
  };
  destination: {
    id: number;
    code: number;
    createdAt: string;
    updatedAt: string;
  };
}

interface PlanProps {
  id: number;
  name: string;
  minutes: number;
  createdAt: string;
  updatedAt: string;
}

interface CalcProps {
  result: {
    priceWithPlan: number;
    priceWithoutPlan: number;
  };
  planName: string;
}

const Home: React.FC = () => {
  const sectionRef = useRef(null);
  const [prices, setPrices] = useState<PriceProps[]>();
  const [plans, setPlans] = useState<PlanProps[]>();
  const [calcValue, setValcValue] = useState<CalcProps>();

  const intersection = useIntersection(sectionRef, {
    root: null,
    rootMargin: '0px',
    threshold: 0.5,
  });

  const fadeIn = (element: any) => {
    gsap.to(element, 1, {
      opacity: 1,
      y: -60,
      ease: 'power4.out',
      stagger: {
        amount: 0.3,
      },
    });
  };

  const fadeOut = (element: any) => {
    gsap.to(element, 1, {
      opacity: 0,
      y: -20,
      ease: 'power4.out',
    });
  };

  /*eslint-disable */
  intersection && intersection.intersectionRatio < 0.5
    ? fadeOut('.fadeIn')
    : fadeIn('.fadeIn');
    /* eslint-enable */

  async function handleFormikSubmit(
    values: {
      minutes: string;
      priceId: string;
      planId: string;
    },
    formikHelpers: FormikHelpers<{
      minutes: string;
      priceId: string;
      planId: string;
    }>
  ) {
    try {
      const { data } = await api.post('/calculate-price', {
        minutes: values.minutes,
        priceId: values.priceId,
        planId: values.planId,
      });

      if (data) {
        setValcValue(data);
      }
    } catch (error) {
      console.log(error);
    }
  }

  async function loadPrices() {
    try {
      const { data } = await api.get('/price');
      if (data) {
        setPrices(data);
      }
    } catch (error) {
      console.log(error);
    }
  }

  async function loadPlans() {
    try {
      const { data } = await api.get('/plan');
      if (data) {
        setPlans(data);
      }
    } catch (error) {
      console.log(error);
    }
  }

  useEffect(() => {
    loadPrices();
    loadPlans();
  }, []);

  return (
    <Container data-test="homeContainer">
      <Parallax
        bgImage={Image1}
        strength={500}
        bgClassName="react-parallax-bgimage"
      >
        <SectionOne data-test="homeSectionOne">
          <h1> Cofira os produtos e serviços da Telzir.</h1>

          <p>
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Animi
            exercitationem provident facilis nulla nam vero quod est magnam
            dolorum itaque ipsum, eligendi quam labore optio explicabo? Ad
            molestiae laboriosam quam?
          </p>
        </SectionOne>
      </Parallax>

      <PlansSection
        bgColor="var(--quaternary)"
        ref={sectionRef}
        data-test="homeSectionTwo"
      >
        <Title gridArea="title" data-test="homeSectionTwoTitle">
          Nossos melhores planos
        </Title>

        <Card
          className="fadeIn"
          bgUrl={phone}
          gridArea="r1"
          data-test="homeSectionTwoCard"
        >
          <CardTitle>Fale Mais 30</CardTitle>

          <CardBody>Fale até 30 minutos sem pagar nada</CardBody>
        </Card>
        <Card
          className="fadeIn"
          bgUrl={phone}
          gridArea="r2"
          data-test="homeSectionTwoCard"
        >
          <CardTitle>Fale Mais 60</CardTitle>

          <CardBody>Fale até 60 minutos sem pagar nada</CardBody>
        </Card>
        <Card
          className="fadeIn"
          bgUrl={phone}
          gridArea="r3"
          data-test="homeSectionTwoCard"
        >
          <CardTitle>Fale Mais 120</CardTitle>

          <CardBody>Fale até 120 minutos sem pagar nada</CardBody>
        </Card>
      </PlansSection>

      <PlansSection bgColor="var(--secondary)" data-test="homeSectionThree">
        <Title gridArea="title">Compare os nosos Planos</Title>
        <Formik
          initialValues={{ minutes: '', priceId: '', planId: '' }}
          validationSchema={validate}
          onSubmit={handleFormikSubmit}
        >
          {({
            errors,
            touched,
            handleChange,
            values,
            handleSubmit,
            isSubmitting,
          }) => (
            <Form onSubmit={handleSubmit} data-test="homeSectionThreeForm">
              <Label>
                <InputTitle htmlFor="priceId">
                  Selecione a origem e destino
                </InputTitle>

                <Select
                  name="priceId"
                  id="priceId"
                  onChange={handleChange}
                  value={values.priceId}
                  error={errors.priceId}
                >
                  <option value="">Origem e Destino</option>
                  {prices &&
                    prices.map((price) => (
                      <option value={price.id} key={price.id}>
                        {`Origem: ${price.source.code}  // Destino: ${price.destination.code}`}
                      </option>
                    ))}
                </Select>
                {errors.priceId && (
                  <Text error={errors.priceId}>{errors.priceId}</Text>
                )}
              </Label>
              <Label>
                <InputTitle htmlFor="planId"> Selecione um plano</InputTitle>

                <Select
                  name="planId"
                  id="planId"
                  onChange={handleChange}
                  value={values.planId}
                  error={errors.planId}
                >
                  <option value="">Plano</option>
                  {plans &&
                    plans.map((plan) => (
                      <option value={plan.id} key={plan.id}>
                        {plan.name}
                      </option>
                    ))}
                </Select>
                {errors.planId && (
                  <Text error={errors.planId}>{errors.planId}</Text>
                )}
              </Label>
              <Label>
                <InputTitle htmlFor="minutes">Quantidade de minutos</InputTitle>
                <Input
                  type="number"
                  id="minutes"
                  min="1"
                  max="1000"
                  name="minutes"
                  onChange={handleChange}
                  value={values.minutes}
                  error={errors.minutes}
                />
                {errors.minutes && (
                  <Text error={errors.minutes}>{errors.minutes}</Text>
                )}
              </Label>
              <ButtonSubmit>Comparar</ButtonSubmit>
            </Form>
          )}
        </Formik>
        {calcValue && (
          <>
            <Card
              color="var(--white)"
              bgColor="var(--primary)"
              gridArea="r2"
              bgUrl={phone2}
              data-test="homeSectionThreeCard"
            >
              <CardTitle>
                Com o plano
                {`${calcValue.planName}`}
              </CardTitle>

              <CardBody>
                R$
                {`${calcValue.result.priceWithPlan}`}
              </CardBody>
            </Card>
            <Card
              color="var(--white)"
              bgColor="var(--primary)"
              gridArea="r3"
              bgUrl={phone2}
              data-test="homeSectionThreeCard"
            >
              <CardTitle>
                Sem o plano
                {`${calcValue.planName}`}
              </CardTitle>

              <CardBody>
                R$
                {`${calcValue.result.priceWithoutPlan}`}
              </CardBody>
            </Card>
          </>
        )}
      </PlansSection>
    </Container>
  );
};

export default Home;
