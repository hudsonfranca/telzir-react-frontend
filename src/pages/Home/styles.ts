import styled from 'styled-components';

interface InputProps {
  error: string | undefined;
}

interface CardProps {
  bgColor?: string;
  color?: string;
  gridArea?: string;
  bgUrl: string;
}

interface TitleProps {
  gridArea?: string;
}

interface TextProps {
  error: string | undefined;
}

export const Container = styled.div`
  width: 100%;
  min-height: 100vh;
`;

export const SectionOne = styled.section`
  width: 100%;
  height: 100vh;
  position: relative;
  z-index: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: var(--white);

  h1 {
    font-size: 4rem;
    padding-bottom: 20px;
    width: 80%;
    text-align: center;
  }

  p {
    font-size: 2rem;
    width: 80%;
    text-align: center;
  }
  &::after {
    content: '';
    width: 100%;
    height: 100%;
    background: rgba(11, 12, 16, 0.5);
    position: absolute;
    top: 0;

    z-index: -1;
  }
`;

export const Card = styled.div<CardProps>`
  width: 100%;
  grid-area: ${({ gridArea }) => gridArea || ''};
  height: 90%;
  background-image: url(${({ bgUrl }) => bgUrl});
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  color: ${({ color }) => color || 'var(--white)'};
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  padding: 20px;
  position: relative;
  z-index: 1;

  &::after {
    content: '';
    width: 100%;
    height: 100%;
    background: rgba(11, 12, 16, 0.7);
    position: absolute;
    top: 0;

    z-index: -1;
  }
`;

export const Title = styled.p<TitleProps>`
  color: var(--white);
  font-size: 4rem;
  height: 100%;
  width: 100%;
  grid-area: ${({ gridArea }) => gridArea || ''};
  text-align: center;
  padding: 50px;
`;

export const CardTitle = styled.p`
  font-size: 3rem;
  height: 100%;
  margin-bottom: 10%;
  text-align: center;
`;

export const CardBody = styled.p`
  font-size: 2rem;
  height: 100%;
  margin-bottom: 30px;
  text-align: center;
`;

export const PlansSection = styled.section<{ bgColor: string }>`
  width: 100%;
  min-height: 100vh;
  background-color: ${({ bgColor }) => bgColor || 'var(--primary)'};
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: auto 1fr;
  grid-template-areas:
    'title title title'
    'r1 r2 r3';
  justify-content: space-between;
  align-items: center;
  padding: 30px;
  gap: 20px;

  @media (max-width: 767px) {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
  }
`;

export const Form = styled.form`
  grid-area: r1;
  width: 100%;
  height: 90%;
  background-color: var(--primary);
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  padding: 10px;

  strong a {
    color: var(--primary);

    :hover,
    :active {
      text-decoration: underline;
    }
  }
`;

export const Label = styled.label`
  display: flex;
  flex-direction: column;
  color: #777;
  font-size: 0.8em;
  margin: 0.5em 0;
  position: relative;
  width: 80%;
  height: 75px;
`;

export const Input = styled.input<InputProps>`
  width: 100%;
  height: 40px;
  border-radius: 5px;
  border: ${({ error }) => (error ? '1px solid var(--red)' : 'none')};
  transition: 0.4s;
  font-size: 1.6rem;
  background-color: var(--white);
  padding-left: 5px;

  :focus {
    outline: none !important;
    border: none;
    box-shadow: ${({ error }) =>
      error ? '0 0 7px  var(--red)' : '0 0 9px var(--hover1)'};
  }
`;

export const Select = styled.select<InputProps>`
  width: 100%;
  height: 40px;
  border: ${({ error }) => (error ? '1px solid var(--red)' : 'none')};
  transition: 0.4s;
  font-size: 1.6rem;
  background-color: var(--white);
  padding-left: 5px;

  :focus {
    outline: none !important;
    border: none;
    box-shadow: ${({ error }) =>
      error ? '0 0 7px  var(--red)' : '0 0 9px var(--quaternary)'};
  }
`;

export const Text = styled.p<TextProps>`
  color: ${({ error }) => (error ? 'var(--red)' : 'var(--white)')};
  width: 80%;
  height: 10px;
  margin-top: 4px;
  font-size: 1.4rem;
`;

export const InputTitle = styled.label`
  color: var(--white);
  font-size: 1.4rem;
  margin-bottom: 5px;
`;

export const ButtonSubmit = styled.button`
  border-radius: 10px;
  width: 80%;
  height: 40px;
  transition: 0.4s;
  color: var(--primary);
  font-size: 1.8rem;
  background-color: var(--quaternary);
`;
