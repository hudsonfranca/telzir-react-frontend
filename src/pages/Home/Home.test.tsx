import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import renderer from 'react-test-renderer';
import Home from './Home';
import 'jest-styled-components';

let wrapper: ShallowWrapper;

beforeEach(() => {
  wrapper = shallow(<Home />);
});

describe('<Home /> redering', () => {
  it('Should render without errors', () => {
    const home = renderer.create(<Home />).toJSON();

    expect(home).toMatchSnapshot();
  });

  it('Should render 1 homeContainer', () => {
    expect(wrapper.find(`[data-test='homeContainer']`).length).toEqual(1);
  });

  it('Should render 1 homeSectionOne', () => {
    expect(wrapper.find(`[data-test='homeSectionOne']`).length).toEqual(1);
  });

  it('Should render 1 homeSectionTwo', () => {
    expect(wrapper.find(`[data-test='homeSectionTwo']`).length).toEqual(1);
  });

  it('Should render 1 homeSectionTwoTitle', () => {
    expect(wrapper.find(`[data-test='homeSectionTwoTitle']`).length).toEqual(1);
  });

  it('Should render 3 homeSectionTwoCard', () => {
    expect(wrapper.find(`[data-test='homeSectionTwoCard']`).length).toEqual(3);
  });

  it('Should render 1 homeSectionThree', () => {
    expect(wrapper.find(`[data-test='homeSectionThree']`).length).toEqual(1);
  });
});
