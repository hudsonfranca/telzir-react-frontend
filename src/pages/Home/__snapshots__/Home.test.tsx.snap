// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> redering Should render without errors 1`] = `
.c0 {
  width: 100%;
  min-height: 100vh;
}

.c1 {
  width: 100%;
  height: 100vh;
  position: relative;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: var(--white);
}

.c1 h1 {
  font-size: 4rem;
  padding-bottom: 20px;
  width: 80%;
  text-align: center;
}

.c1 p {
  font-size: 2rem;
  width: 80%;
  text-align: center;
}

.c1::after {
  content: '';
  width: 100%;
  height: 100%;
  background: rgba(11,12,16,0.5);
  position: absolute;
  top: 0;
  z-index: -1;
}

.c4 {
  width: 100%;
  grid-area: r1;
  height: 90%;
  background-image: url(phone.jpg);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  color: var(--white);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
  position: relative;
  z-index: 1;
}

.c4::after {
  content: '';
  width: 100%;
  height: 100%;
  background: rgba(11,12,16,0.7);
  position: absolute;
  top: 0;
  z-index: -1;
}

.c7 {
  width: 100%;
  grid-area: r2;
  height: 90%;
  background-image: url(phone.jpg);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  color: var(--white);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
  position: relative;
  z-index: 1;
}

.c7::after {
  content: '';
  width: 100%;
  height: 100%;
  background: rgba(11,12,16,0.7);
  position: absolute;
  top: 0;
  z-index: -1;
}

.c8 {
  width: 100%;
  grid-area: r3;
  height: 90%;
  background-image: url(phone.jpg);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  color: var(--white);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
  position: relative;
  z-index: 1;
}

.c8::after {
  content: '';
  width: 100%;
  height: 100%;
  background: rgba(11,12,16,0.7);
  position: absolute;
  top: 0;
  z-index: -1;
}

.c3 {
  color: var(--white);
  font-size: 4rem;
  height: 100%;
  width: 100%;
  grid-area: title;
  text-align: center;
  padding: 50px;
}

.c5 {
  font-size: 3rem;
  height: 100%;
  margin-bottom: 10%;
  text-align: center;
}

.c6 {
  font-size: 2rem;
  height: 100%;
  margin-bottom: 30px;
  text-align: center;
}

.c2 {
  width: 100%;
  min-height: 100vh;
  background-color: var(--quaternary);
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: auto 1fr;
  grid-template-areas: 'title title title' 'r1 r2 r3';
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 30px;
  gap: 20px;
}

.c9 {
  width: 100%;
  min-height: 100vh;
  background-color: var(--secondary);
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: auto 1fr;
  grid-template-areas: 'title title title' 'r1 r2 r3';
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 30px;
  gap: 20px;
}

.c10 {
  grid-area: r1;
  width: 100%;
  height: 90%;
  background-color: var(--primary);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
}

.c10 strong a {
  color: var(--primary);
}

.c10 strong a:hover,
.c10 strong a:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: #777;
  font-size: 0.8em;
  margin: 0.5em 0;
  position: relative;
  width: 80%;
  height: 75px;
}

.c14 {
  width: 100%;
  height: 40px;
  border-radius: 5px;
  border: none;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  font-size: 1.6rem;
  background-color: var(--white);
  padding-left: 5px;
}

.c14:focus {
  outline: none !important;
  border: none;
  box-shadow: 0 0 9px var(--hover1);
}

.c13 {
  width: 100%;
  height: 40px;
  border: none;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  font-size: 1.6rem;
  background-color: var(--white);
  padding-left: 5px;
}

.c13:focus {
  outline: none !important;
  border: none;
  box-shadow: 0 0 9px var(--quaternary);
}

.c12 {
  color: var(--white);
  font-size: 1.4rem;
  margin-bottom: 5px;
}

.c15 {
  border-radius: 10px;
  width: 80%;
  height: 40px;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  color: var(--primary);
  font-size: 1.8rem;
  background-color: var(--quaternary);
}

@media (max-width:767px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:767px) {
  .c9 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div
  className="c0"
  data-test="homeContainer"
>
  <div
    className="react-parallax "
    style={
      Object {
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <img
      alt=""
      className="react-parallax-bgimage"
      src="home.jpg"
      style={
        Object {
          "MozBackfaceVisibility": "hidden",
          "MsBackfaceVisibility": "hidden",
          "WebkitBackfaceVisibility": "hidden",
          "WebkitTransform": "translate3d(-50%, 0, 0)",
          "WebkitTransformStyle": "preserve-3d",
          "left": "50%",
          "position": "absolute",
          "transform": "translate3d(-50%, 0, 0)",
        }
      }
    />
    <div
      className="react-parallax-content"
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <section
        className="c1"
        data-test="homeSectionOne"
      >
        <h1>
           Cofira os produtos e serviços da Telzir.
        </h1>
        <p>
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Animi exercitationem provident facilis nulla nam vero quod est magnam dolorum itaque ipsum, eligendi quam labore optio explicabo? Ad molestiae laboriosam quam?
        </p>
      </section>
    </div>
  </div>
  <section
    className="c2"
    data-test="homeSectionTwo"
  >
    <p
      className="c3"
      data-test="homeSectionTwoTitle"
    >
      Nossos melhores planos
    </p>
    <div
      className="c4 fadeIn"
      data-test="homeSectionTwoCard"
    >
      <p
        className="c5"
      >
        Fale Mais 30
      </p>
      <p
        className="c6"
      >
        Fale até 30 minutos sem pagar nada
      </p>
    </div>
    <div
      className="c7 fadeIn"
      data-test="homeSectionTwoCard"
    >
      <p
        className="c5"
      >
        Fale Mais 60
      </p>
      <p
        className="c6"
      >
        Fale até 60 minutos sem pagar nada
      </p>
    </div>
    <div
      className="c8 fadeIn"
      data-test="homeSectionTwoCard"
    >
      <p
        className="c5"
      >
        Fale Mais 120
      </p>
      <p
        className="c6"
      >
        Fale até 120 minutos sem pagar nada
      </p>
    </div>
  </section>
  <section
    className="c9"
    data-test="homeSectionThree"
  >
    <p
      className="c3"
    >
      Compare os nosos Planos
    </p>
    <form
      className="c10"
      data-test="homeSectionThreeForm"
      onSubmit={[Function]}
    >
      <label
        className="c11"
      >
        <label
          className="c12"
          htmlFor="priceId"
        >
          Selecione a origem e destino
        </label>
        <select
          className="c13"
          id="priceId"
          name="priceId"
          onChange={[Function]}
          value=""
        >
          <option
            value=""
          >
            Origem e Destino
          </option>
        </select>
      </label>
      <label
        className="c11"
      >
        <label
          className="c12"
          htmlFor="planId"
        >
           Selecione um plano
        </label>
        <select
          className="c13"
          id="planId"
          name="planId"
          onChange={[Function]}
          value=""
        >
          <option
            value=""
          >
            Plano
          </option>
        </select>
      </label>
      <label
        className="c11"
      >
        <label
          className="c12"
          htmlFor="minutes"
        >
          Quantidade de minutos
        </label>
        <input
          className="c14"
          id="minutes"
          max="1000"
          min="1"
          name="minutes"
          onChange={[Function]}
          type="number"
          value=""
        />
      </label>
      <button
        className="c15"
      >
        Comparar
      </button>
    </form>
  </section>
</div>
`;
